### PR TITLE
[feat][elasticsearch-sink] Option to strip out non printable characters

### PR DESCRIPTION
**Pull request by nicoloboschi**
### Motivation

If the message value contains non-printable characters you will get 

```
2022-04-22T22:37:29.673384094Z 22:37:29.668 [tenant/ns/topic] ERROR org.apache.pulsar.io.elasticsearch.ElasticSearchSink - Malformed document messageId=73895:0:1
2022-04-22T22:37:29.673415795Z com.fasterxml.jackson.core.JsonParseException: Illegal character ((CTRL-CHAR, code 0)): only regular white space (\r, \n, \t) is allowed between tokens
2022-04-22T22:37:29.673424695Z  at [Source: (String)"\u0000\u0000\u0002�\u000C\u0002\u001Epick_start_time\u0000\u0002\u00081211\u0002\u000ESafeway\u0002�����_\u0000\u0000\u0002\u00021\u0002\u0018445184429011"; line: 1, column: 2]
```

Even if you set malformedDocAction to IGNORE, the message will be re-delivered. In case of KEY_SHARED subscriptions this will lead to stuck subscriptions scenario.

The issue is that [JSON format doesn't accept this kind of characters](https://datatracker.ietf.org/doc/html/rfc8259#section-7). 

>All Unicode characters may be placed within the
   quotation marks, except for the characters that MUST be escaped:
   quotation mark, reverse solidus, and the control characters (U+0000
   through U+001F).

Since usually these characters are useless, it is better to drop them all instead of encoding (which is not simple because it depends how much the json is malformed. For example, inside a key or a value you can encode them but you cannot between tokens)

### Modifications
- New option `stripNonPrintableCharacters` default=true (which will trigger a different behaviour by default) which removes the non printable characters from the output json (only for the document, not the _id because ElasticSearch doesn't care if the _id a valid json or not). The stripping is done via RegEx because, unfortunately, Jackson Mapper doesn't support this out of the box.

- [x] `doc`

### CODE REVIEW

File: site2/docs/io-elasticsearch-sink.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ The configuration of the Elasticsearch sink connector has the following properti
 | `compatibilityMode` | enum (AUTO,ELASTICSEARCH,ELASTICSEARCH_7,OPENSEARCH) | AUTO |  | Specify compatibility mode with the ElasticSearch cluster. `AUTO` value will try to auto detect the correct compatibility mode to use. Use `ELASTICSEARCH_7` if the target cluster is running ElasticSearch 7 or prior. Use `ELASTICSEARCH` if the target cluster is running ElasticSearch 8 or higher. Use `OPENSEARCH` if the target cluster is running OpenSearch. |
 | `token` | String| false | " " (empty string)|The token used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
 | `apiKey` | String| false | " " (empty string)|The apiKey used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
+| `stripNonPrintableCharacters` | Boolean| false | true|If stripNonPrintableCharacters is true, all non-printable characters will be removed from the document. |
```
Comment (Anonymitaet) [inline site2/docs/io-elasticsearch-sink.md:83]:
```suggestion
| `stripNonPrintableCharacters` | Boolean| false | true|If `stripNonPrintableCharacters` is true, all non-printable characters are removed from the document. |
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true.

https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

**Comment by Anonymitaet** [comment]
cc @momo-jun

File: site2/docs/io-elasticsearch-sink.md [hunk 2]
```diff
@@ -80,6 +80,7 @@ The configuration of the Elasticsearch sink connector has the following properti
 | `compatibilityMode` | enum (AUTO,ELASTICSEARCH,ELASTICSEARCH_7,OPENSEARCH) | AUTO |  | Specify compatibility mode with the ElasticSearch cluster. `AUTO` value will try to auto detect the correct compatibility mode to use. Use `ELASTICSEARCH_7` if the target cluster is running ElasticSearch 7 or prior. Use `ELASTICSEARCH` if the target cluster is running ElasticSearch 8 or higher. Use `OPENSEARCH` if the target cluster is running OpenSearch. |
 | `token` | String| false | " " (empty string)|The token used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
 | `apiKey` | String| false | " " (empty string)|The apiKey used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
+| `stripNonPrintableCharacters` | Boolean| false | true|If `stripNonPrintableCharacters` is true, all non-printable characters are removed from the document. |
```
Comment (momo-jun) [inline site2/docs/io-elasticsearch-sink.md:83]:
```suggestion
| `stripNonPrintableCharacters` | Boolean| false | true| Whether to remove all non-printable characters from the document or not. If it is set to true, all non-printable characters are removed from the document. |
```

**Review by eolivelli — APPROVED** [review]

**Pull request merged**